contracts/contracts/harvest/Harvester.sol

```diff
@@ -136,16 +156,17 @@ contract Harvester is Governable {
             "Harvest reward fee should not be over 10%"
         );
         require(
-            _uniswapV2CompatibleAddr != address(0),
-            "Uniswap compatible address should be non zero address"
+            (_uniswapV2CompatibleAddr != address(0)) == (_v3Fee == 0),
```

Comment:
💪

Reply:
I am impressed. Worked it out on paper to verify.

Basically this is doing an XOR so that one and only one of the two conditions is true. 

It's a little confusing to understand because one side is true when set, and the other side is false when set. I think it would be clearer if this used the normal XOR substitute "!=" for the middle comparison operator, and then did `(_v3Fee > 0)`. This way both sides of the check are the same kind, with true for this kind is set, and false for this kind is not set.

Thanks, I learned something today.